add rocketkeyboards mercury65

drewpyun opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->
added QMK and VIA keymaps for the mercury65 keyboard.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

zvecr reviewed (changes requested):
Per the PR checklist,

* config.h, rules.mk and layout macro in mercury65.h should be migrated to info.json (<https://docs.qmk.fm/#/reference_info_json>)
* Providing only a `LAYOUT_all` layout macro is invalid as can be seen in the lint output

drashna commented:
`qmk info -f json` can help with migrating the settings

drewpyun commented:
Updated info.json, removed config.h, rules.mk and layout macro in mercury65.h as per the requested changes. 
Thank you for the qmk info -f json command.

waffle87 commented on keyboards/drewkeys/mercury65/info.json:
```diff
@@ -0,0 +1,109 @@
+    "bootloader": "atmel-dfu",
+    "diode_direction": "COL2ROW",
+    "url": "https://www.rocketkeyboards.com/collections/mercury65/products/gb-mercury65-keyboard",
+    "keyboard_folder": "drewkeys/mercury65",
```
```suggestion
```

waffle87 commented on keyboards/drewkeys/mercury65/info.json:
```diff
@@ -0,0 +1,109 @@
+        "pid": "0x1034",
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "audio": false,
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": false
+    },
+    "matrix_size": {
+       "cols": 17,
+        "rows": 5
+    },
```
```suggestion
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true
    },
```

waffle87 commented on keyboards/drewkeys/mercury65/keymaps/default/readme.md:
This file should be removed.

waffle87 commented on keyboards/drewkeys/mercury65/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,63 @@
+// entirely and just use numbers.
+enum layer_names {
+    _BASE,
+    _FN1,
+    _FN2,
+    _FN3
```
```suggestion
    _FN1
```

waffle87 commented on keyboards/drewkeys/mercury65/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,63 @@
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,          _______,
+        _______, _______, _______,                            _______,                            _______, _______,          _______, _______, _______
+    ),
+    
+    [_FN2] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,          _______,
+        _______, _______, _______,                            _______,                            _______, _______,          _______, _______, _______
+    ),
+
+    [_FN3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,          _______,
+        _______, _______, _______,                            _______,                            _______, _______,          _______, _______, _______
+    ),
+
```
```suggestion
    )
```

waffle87 commented on keyboards/drewkeys/mercury65/keymaps/via/readme.md:
This file should be removed.

waffle87 commented on keyboards/drewkeys/mercury65/readme.md:
```diff
@@ -0,0 +1,22 @@
+# Mercury65
+
+![Mercury65](https://i.imgur.com/rjfARj3.jpeg)
```
```suggestion
![Mercury65](https://i.imgur.com/rjfARj3h.jpeg)
```

waffle87 commented on keyboards/drewkeys/mercury65/rules.mk:
```diff
@@ -0,0 +1 @@
+# all settings within info.json
```
```suggestion
# This file intentionally left blank
```

waffle87 commented on keyboards/drewkeys/mercury65/readme.md:
```diff
@@ -0,0 +1,22 @@
+Flashing example for this keyboard:
+
+    make drewkeys/mercury65:default:flash
+
```
```suggestion
## Bootloader

```

drewpyun commented:
Thanks for the review.
